### PR TITLE
BESman Utility PR

### PR DESCRIPTION
**Pull request by senthilbk**
This PR is created to incorporated changes from my repo to base repo

### CODE REVIEW

File: bin/release.sh [hunk 1]
```diff
@@ -1,27 +1,59 @@
...
+#setting up environment variables
+if [[ -z $BES_ARCHIVE_DOWNLOAD_REPO ]];
+    then
+        BES_ARCHIVE_DOWNLOAD_REPO={BES_ARCHIVE_DOWNLOAD_REPO:-BESman}
+fi
+
+if [[ -z $BESMAN_NAMESPACE ]];
+    then
+        #BESMAN_NAMESPACE={BESMAN_NAMESPACE:-hyperledgerkochi}
+	BESMAN_NAMESPACE={BESMAN_NAMESPACE:-senthilbk}
```
Comment (asa1997) [inline bin/release.sh:23]: Please make the default namespace as hyperledgerkochi.

File: dist/get.besman.io [hunk 2]
```diff
@@ -0,0 +1,353 @@
+#!/bin/bash
+
+#Install: stable
+
+# Global variables
+BESMAN_PLATFORM=$(uname)
+export BESMAN_SERVICE="https://raw.githubusercontent.com"
+
+#BESMAN_NAMESPACE="hyperledgerkochi"
+BESMAN_NAMESPACE="senthilbk"
```
Comment (asa1997) [inline dist/get.besman.io:10]: Please make the namespace as hyperledgerkochi.

File: dist/get.besman.io [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+	besman_zip_file=$(cygpath -w "$besman_zip_file")
+	besman_stage_folder=$(cygpath -w "$besman_stage_folder")
+fi
+unzip -qo "$besman_zip_file" -d "$besman_stage_folder"
+
+
+echo "Install scripts..."
+
+
+curl -sL "https://raw.githubusercontent.com/${BESMAN_NAMESPACE}/BESman/dev/dist/environments" > tmp.txt
```
Comment (asa1997) [inline dist/get.besman.io:254]: Please make the branch as master.

File: scripts/get.besman.io [hunk 4]
```diff
@@ -0,0 +1,353 @@
+#!/bin/bash
```
Comment (asa1997) [inline scripts/get.besman.io:1]: This file should not be present under the master branch. Only the template for the same should be present.

File: scripts/tmpl/get.besman.io.tmpl [hunk 5]
```diff
@@ -0,0 +1,346 @@
+#!/bin/bash
+
+#Install: stable
+
+# Global variables
+BESMAN_PLATFORM=$(uname)
+export BESMAN_SERVICE="https://raw.githubusercontent.com"
+
+#BESMAN_NAMESPACE="hyperledgerkochi"
+BESMAN_NAMESPACE="senthilbk"
```
Comment (asa1997) [inline scripts/tmpl/get.besman.io.tmpl:10]: Please make the value of the namespace as a variable like in KOBman.

File: scripts/tmpl/get.besman.io.tmpl [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+BESMAN_VERSION="0.0.1"
```
Comment (asa1997) [inline scripts/tmpl/get.besman.io.tmpl:11]: Please take out the line.

File: scripts/tmpl/get.besman.io.tmpl [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+	besman_zip_file=$(cygpath -w "$besman_zip_file")
+	besman_stage_folder=$(cygpath -w "$besman_stage_folder")
+fi
+unzip -qo "$besman_zip_file" -d "$besman_stage_folder"
+
+
+echo "Install scripts..."
+
+
+curl -sL "https://raw.githubusercontent.com/${BESMAN_NAMESPACE}/BESman/dev/dist/environments" > tmp.txt
```
Comment (asa1997) [inline scripts/tmpl/get.besman.io.tmpl:247]: Please change from dev branch to master branch.

File: scripts/tmpl/get.besman.io.tmpl [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+echo "besman_beta_channel=false" >> "$besman_config_file"
+echo "besman_debug_mode=true" >> "$besman_config_file"
+echo "besman_colour_enable=true" >> "$besman_config_file"
+
+echo "Setting up user configs"
+touch "$besman_user_config_file"
+echo "BESMAN_VERSION=$BESMAN_VERSION" >> "$besman_user_config_file"
+echo "BESMAN_USER_NAMESPACE=" >> "$besman_user_config_file"
+echo "BESMAN_ENV_ROOT=$HOME/BESman_env" >> "$besman_user_config_file"
+echo "BESMAN_NAMESPACE=senthilbk" >> "$besman_user_config_file"
```
Comment (asa1997) [inline scripts/tmpl/get.besman.io.tmpl:211]: Please change the value as hyperledgerkochi

File: scripts/tmpl/get.besman.io.tmpl [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+echo "besman_beta_channel=false" >> "$besman_config_file"
+echo "besman_debug_mode=true" >> "$besman_config_file"
+echo "besman_colour_enable=true" >> "$besman_config_file"
+
+echo "Setting up user configs"
+touch "$besman_user_config_file"
+echo "BESMAN_VERSION=$BESMAN_VERSION" >> "$besman_user_config_file"
+echo "BESMAN_USER_NAMESPACE=" >> "$besman_user_config_file"
+echo "BESMAN_ENV_ROOT=$HOME/BESman_env" >> "$besman_user_config_file"
+echo "BESMAN_NAMESPACE=Be-Secure" >> "$besman_user_config_file"
```
Comment (asa1997) [inline scripts/tmpl/get.besman.io.tmpl:215]: Please use the variable instead of hard coding the value.

File: src/main/bash/besman-init.sh [hunk 10]
```diff
@@ -1,41 +1,56 @@
...
-#   limitations under the License.
-#
-
+function __besman_set_user_configs
+{
+	if [[ -f $BESMAN_DIR/etc/user-config.cfg ]]; then
+		source $BESMAN_DIR/etc/user-config.cfg
+	fi
+# The functions sets all the user configs specified in the user-config.cfg file
+	if [[ ! -f $HOME/.besman/etc/user-config.cfg ]]; then
```
Comment (asa1997) [inline src/main/bash/besman-init.sh:9]: Move this if else statement to line 5.

File: src/main/bash/besman-init.sh [hunk 11]
```diff
@@ -1,41 +1,56 @@
...
+		return 1
+	else
+		source $HOME/.besman/etc/user-config.cfg
```
Comment (asa1997) [inline src/main/bash/besman-init.sh:12]: Please use the env variable for .besman path.

File: src/main/bash/besman-init.sh [hunk 12]
```diff
@@ -1,41 +1,56 @@
...
-#   Unless required by applicable law or agreed to in writing, software
-#   distributed under the License is distributed on an "AS IS" BASIS,
-#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#   See the License for the specific language governing permissions and
-#   limitations under the License.
-#
-
+function __besman_set_user_configs
+{
+	if [[ -f $BESMAN_DIR/etc/user-config.cfg ]]; then
```
Comment (asa1997) [inline src/main/bash/besman-init.sh:5]: You may take out this if statement as it is covered in the one below.

File: src/main/bash/besman-init.sh [hunk 13]
```diff
@@ -1,41 +1,56 @@
...
+		if echo $user_configs | grep -q "^#"
+			then
+				continue
+		fi
+		echo $user_configs > $HOME/tmp.txt
+		local user_config_param=$(cut -d "=" -f 1 $HOME/tmp.txt)
+		local user_config_values=$(cut -d "=" -f 2 $HOME/tmp.txt)
+		unset $user_config_param
+		export $user_config_param=$user_config_values
+	done < $HOME/.besman/etc/user-config.cfg
```
Comment (asa1997) [inline src/main/bash/besman-init.sh:24]: Please use the env variable here.

File: src/main/bash/besman-main.sh [hunk 14]
```diff
@@ -1,152 +1,109 @@
...
-	#
-
-	# Check version and candidates cache
-	if [[ "$COMMAND" != "update" ]]; then
-		___besman_check_candidates_cache "$BESMAN_CANDIDATES_CACHE" || return 1
-		___besman_check_version_cache
+function bes {
+	[[ -z "$1" ]] && __bes_help && return 0
+
+	function __besman_check_for_env_file
```
Comment (asa1997) [inline src/main/bash/besman-main.sh:6]: Please move this function outside "function bes". Trying placing this function after "bes" func. If it doesnt work move this function to the top.

File: src/main/bash/besman-main.sh [hunk 15]
```diff
@@ -1,152 +1,109 @@
...
+		if [[ ! -f $BESMAN_DIR/envs/besman-$environment.sh ]]; then
+			__besman_echo_red "Wrong Command Format"
+			__besman_echo_red "Could not find file besman-$environment.sh"
+			__besman_echo_white "Make sure you have given the correct environment name"
+			__besman_echo_white "If the issue persists, re-install BESman and try again"
+			return 1
+		fi
+	}
+
+	function __besman_check_for_command_file
```
Comment (asa1997) [inline src/main/bash/besman-main.sh:18]: Please move this function outside "function bes". Trying placing this function after "bes" func. If it doesnt work move this function to the top.

File: src/main/bash/commands/besman-update.sh [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+                        if [[ -n $diff_remote ]];then
+                                __besman_echo_no_colour "" >> $cached_list
+                                for i in ${diff_remote[@]};do
+                                        environment_name=$(echo $i | cut -d "/" -f 3 | cut -d "," -f 1)
+                                        version_name=$(echo $i | cut -d "," -f 2)
+                                        #grep -v "$environment_name" $cached_list >$HOME/tmpfile && mv $HOME/tmpfile $cached_list
+                                        grep -v "$i" $cached_list >$HOME/tmpfile && mv $HOME/tmpfile $cached_list
+                                        flag=2
+                                        # Since there is difference between remote repo and local repo, Respective envrioment files and foler will be uninstalled
+                                        __bes_uninstall $environment_name $version_name
```
Comment (asa1997) [inline src/main/bash/commands/besman-update.sh:70]: I dont think you actually need to call the uninstall function. You can just remove the files here.

**Comment by asa1997** [comment]
@senthilbk Please address the review comments.

File: dist/environments [hunk 17]
```diff
@@ -0,0 +1 @@
+BESman
```
Comment (asa1997) [inline dist/environments:1]: BESman -> BeSman

File: dist/list.txt [hunk 18]
```diff
@@ -0,0 +1 @@
+BESman,0.0.1
```
Comment (asa1997) [inline dist/list.txt:1]: BeSman

Comment (asa1997) [reply dist/list.txt:1]: Please take a look at other files and change the utility's name to BeSman if needed.

**Comment by asa1997** [comment]
@senthilbk @panickervinod @sumodgeorge LGTM
@panickervinod Please merge this pull request.

**Comment by panickervinod** [comment]
@asa1997 Please do confirm post this merge.

**Pull request merged**